cmd/kpod/pull.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	cp "github.com/containers/image/copy"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/signature"
+	is "github.com/containers/image/storage"
+	"github.com/containers/image/transports/alltransports"
+	"github.com/containers/image/types"
+	"github.com/containers/storage"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const (
+	// DefaultRegistry is a prefix that we apply to an image name
+	// to check docker hub first for the image
+	DefaultRegistry = "docker://"
+)
+
+// PullOptions is used to initialize and store the information received
+// from the command line
+type PullOptions struct {
+	FromImage           string
+	AllTags             bool
+	DisableContentTrust bool
+}
+
+var (
+	pullFlags = []cli.Flag{
+	// cli.BoolFlag{
+	// 	Name:  "all-tags, a",
+	// 	Usage: "Download all tagged images in the repository",
+	// },
+	// cli.BoolFlag{
+	// 	Name:  "disable-content-trust",
+	// 	Usage: "Skip image verification",
+	// },
+	}
+
+	pullDescription = "Pulls an image from a registry and stores it locally.\n" +
+		"An image can be pulled using its tag or digest. If a tag is not\n" +
+		"specified, the image with the 'latest' tag (if it exists) is pulled."
+	pullCommand = cli.Command{
+		Name:        "pull",
+		Usage:       "pull an image from a registry",
+		Description: pullDescription,
+		Flags:       pullFlags,
+		Action:      pullCmd,
+		ArgsUsage:   "",
+	}
+)
+
+// pullCmd gets the data from the command line and calls pullImage
+// to copy an image from a registry to a local machine
+func pullCmd(c *cli.Context) error {
+
+	args := c.Args()
+	if len(args) == 0 {
+		logrus.Errorf("an image name must be specified")
+		return nil
+	}
+	if len(args) > 1 {
+		logrus.Errorf("too many arguments. Requires exactly 1")
+		return nil
+	}
+	image := args[0]
+
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	allTags := false
+	// if c.IsSet("all-tags") {
```

Comment:
I have this uncommented because it is a member of the pullOptions struct, which I pass to the pullImage function. I have the flags commented out so that it doesn't appear in help or auto-complete.